main.tf

```
@@ -6,8 +6,8 @@ locals {
 
 data "aws_ami" "eks_ami" {
   most_recent = true
-  name_regex  = var.ami_regex_override == "" ? "^bottlerocket-aws-k8s-1.27-x86_64-v1.17.0" : var.ami_regex_override
-  owners      = ["092701018921"]
+  name_regex  = var.use_bottlerocket ? "^bottlerocket-aws-k8s-${var.cluster_version}-x86_64-v1.17.0" : (var.ami_regex_override == "" ? "^amzn2-eks-${var.cluster_version}-gi-${var.ami_date}" : var.ami_regex_override)
+  owners      = var.use_bottlerocket ? ["092701018921"] : (length(var.ami_owner_override) > 0 && var.ami_owner_override[0] != "" ? var.ami_owner_override : ["743302140042"])
```

The var should be converted to a string, not a list.  then this logic becomes much more clear.  I also moved the override to be the first priority.
```suggestion
  # If an ami_owner_override is provided, use it.  Otherwise use the AWS AMI's for bottlerocket, and CMS AMIs AL2
  owners      = var.ami_owner_override != "" ? [var.ami_owner_override] : (var.use_bottlerocket ? ["092701018921"] : ["743302140042"])
```

---

not resolved yet